src/doc/languagespec.tex

```diff
@@ -2812,56 +2821,23 @@ \section{Global variables}
 \end{table}
 
 
-\begin{comment}  % old table
-\begin{table}
-\begin{tabular}{|p{1in}p{0.75in}p{0.75in}p{0.75in}p{1.7in}|}
-\hline
-{\bf Variable} & surface & displacement & volume & light \\
-\hline
-\P            & R     & RW & R     & R \\
-\I            & R     &    & R &   \\
-\N            & RW    & RW &  &  \\
-{\cf Ng}      & R     & R  &  &  \\
-{\cf dPdu}    & R     & R  &  &  \\
-{\cf dPdv}    & R     & R  &  &  \\
-{\cf u, v}    & R     & R  & R$^1$ & R$^1$ \\
-{\cf time}    & R     & R  & R     & R \\
-{\cf dtime}   & R     & R  & R     & R \\
-{\cf dPdtime} & R     & R  & R     & R \\
-\Ci           & RW    &    & RW    & \\
-%\Oi           & RW    &    & RW    & \\
-%{\cf L}       & R$^2$ &    & R$^2$ & RW \\
-{\cf L}       &       &    &       & RW \\
-%{\cf Cl}      & R$^2$ &    & R$^2$ & RW \\
-{\cf Cl}      &       &    &       & RW \\
-{\cf Ps}      &       &    &       & R \\
-{\cf Ns}      &       &    &       & R \\
-%{\cf Pl}      &       &    & R$^2$ & R \\
-%{\cf Nl}      &       &    & R$^2$ & R \\
-{\cf Pl}      &       &    &  & R \\
-{\cf Nl}      &       &    &  & R \\
-\hline 
-\end{tabular}
-\end{table}
-\end{comment}
-
 \begin{table}
 \begin{tabular}{|p{1in}p{0.75in}p{0.75in}p{0.75in}|}
 \hline
-{\bf Variable} & surface & displacement & volume  \\
+{\bf Variable} & surface & displacement & volume & camera & image \\
 \hline
-\P            & R     & RW & R      \\
-\I            & R     &    & R    \\
-\N            & RW    & RW &    \\
-{\cf Ng}      & R     & R  &    \\
-{\cf dPdu}    & R     & R  &    \\
-{\cf dPdv}    & R     & R  &    \\
-{\cf Ps}      &       &    & R   \\
-{\cf u, v}    & R     & R  & R   \\
-{\cf time}    & R     & R  & R       \\
-{\cf dtime}   & R     & R  & R       \\
-{\cf dPdtime} & R     & R  & R       \\
-\Ci           & RW    &    & RW     \\
+\P            & R        & RW           & R      & RW     &       \\
+\I            & R        &              & R      & RW     &       \\
+\N            & RW       & RW           &        &        &       \\
+{\cf Ng}      & R        & R            &        &        &       \\
+{\cf dPdu}    & R        & R            &        & R      &       \\
+{\cf dPdv}    & R        & R            &        & R      &       \\
+%{\cf Ps}      &          &              & R      &        &       \\
+{\cf u, v}    & R        & R            & R      & R      & R     \\
+{\cf time}    & R        & R            & R      & R      & R     \\
+{\cf dtime}   & R        & R            & R      & R      & R     \\
+{\cf dPdtime} & R        & R            & R      & R      & R     \\
```

Comment:
I don't think `time`, `dtime`  or `dPdtime` are relevant to the image shaders (for now at least).

Reply:
yeah, you're probably right

Reply:
ok, dPdtime, certainly. But might time and dtime (for image shaders) be the frame time and duration? I can imagine wanting time-varying effects implemented in these shaders, for example.

Reply:
Makes sense. Maybe we should fix our renderer to provide `time` as `frame_time+shutter_sample` for the other shader types then.